src/Npgsql/Replication/ReplicationConnection.cs

```diff
@@ -432,7 +427,9 @@ async Task<TimelineHistoryFile> TimelineHistoryInternal()
                 }
 
                 var buf = connector.ReadBuffer;
-                var columnStream = new NpgsqlReadBuffer.ColumnStream(buf);
+
+                // Cancellation is handled at the replication level - we don't want every ReadAsync
+                var columnStream = new NpgsqlReadBuffer.ColumnStream(connector, startCancellableOperations: false);
```

Comment:
@vonzshik FYI this is the one possible exception to the rule about user actions with tokens not having nested operations... The entire replication is a user action with a cancellation token, but if the user is doing physical replication (not logical), they get a Stream back which they can read from (sequentially); so ReadAsync on that may accept cancellation tokens.

This is the only case I'm aware of in Npgsql where it may make sense to have two linked cancellation tokens when reading. As this is a really really niche edge case, I've simply hacked it to ignore the token passed to the stream - you can only cancel via the token passed when starting replication.